XenAdmin/Alerts/Types/ClientUpdateAlert.cs

```diff
@@ -106,15 +107,24 @@ public override bool Equals(Alert other)
 
         public static void DownloadAndInstallNewClient(ClientUpdateAlert updateAlert, IWin32Window parent)
         {
-            using (var dialog = new WarningDialog(
-                           string.Format(Messages.UPDATE_CLIENT_CONFIRMATION_MESSAGE, BrandManager.BrandConsole),
-                           ThreeButtonDialog.ButtonOK, ThreeButtonDialog.ButtonCancel)
-                       {WindowTitle = string.Format(Messages.UPDATE_CLIENT_CONFIRMATION_MESSAGE_TITLE, BrandManager.BrandConsole)})
+            bool currentTasks = false;

```

Comment:
If we went with this solution, we would also need to modify slightly the blurb and buttons of the CloseXenCenterWarningDialog to mention that we're exiting because we're updating.

However, a more important issue I see is that launching this dialog should be the last thing to happen before exiting, which is not the case because we haven't downloaded the update yet. On a slow network this might take a while and it's not certain that no other actions will start in the meantime, plus that during that interval the user might have the time to wait for their actions to finish instead of cancelling them (even if no one is forcing them to do so).

Maybe we should leave the original warning dialog as it was, and if we make the change I'm mentioning in the other comment, check and launch the CloseXenCenterWarningDialog (and then we don't need to change any wording) in that block immediately before calling the msi process as a last line of defence.

Reply:
> Maybe we should leave the original warning dialog as it was, and if we make the change I'm mentioning in the other comment, check and launch the CloseXenCenterWarningDialog (and then we don't need to change any wording) in that block immediately before calling the msi process as a last line of defence.

I agree that this is the best option, as users might find frustrating having to start over just because they forgot to close a couple of actions.